api/hypershift/v1beta1/azure_types.go

```diff
@@ -1,144 +1,177 @@
-	// VMSize is the Azure VM instance type to use for the nodes being created in the nodepool.
+	// vmSize is the Azure VM instance type to use for the nodes being created in the nodepool.
 	//
+	// TODO: This should be validated with a regex, based on https://learn.microsoft.com/en-us/azure/virtual-machines/vm-naming-conventions
 	// +kubebuilder:validation:Required
-	VMSize string `json:"vmsize"`
+	VMSize string `json:"vmSize"`
 
-	// ImageID is the id of the image to boot from. If unset, the default image at the location below will be used and
+	// image is used to configure the VM boot image. If unset, the default image at the location below will be used and
 	// is expected to exist: subscription/<subscriptionID>/resourceGroups/<resourceGroupName>/providers/Microsoft.Compute/images/rhcos.x86_64.vhd.
 	// The <subscriptionID> and the <resourceGroupName> are expected to be the same resource group documented in the
 	// Hosted Cluster specification respectively, HostedCluster.Spec.Platform.Azure.SubscriptionID and
 	// HostedCluster.Spec.Platform.Azure.ResourceGroupName.
+	// The image is immutable once set.
 	//
-	// +kubebuilder:validation:XValidation:rule="self == oldSelf", message="Image is immutable"
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf", message="image is immutable"
 	// +kubebuilder:validation:Required
 	Image AzureVMImage `json:"image"`
 
-	// DiskSizeGB is the size in GB to assign to the OS disk
-	// CAPZ default is 30GB, https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/b3708019a67ff19407b87d63c402af94ca4246f6/api/v1beta1/types.go#L599
+	// diskSizeGB is the size in GB to assign to the OS disk. This should be between 16 and <TODO find a limit>.
+	// When not set, this means no opinion and the platform is left to choose a reasonable default, which is subject to change over time.
+	// The current default is 30.
+	// TODO: Is this GB or GiB?
 	//
-	// +kubebuilder:default:=30
 	// +kubebuilder:validation:Minimum=16
 	// +optional
 	DiskSizeGB int32 `json:"diskSizeGB,omitempty"`
 
-	// DiskStorageAccountType is the disk storage account type to use. Valid values are:
-	// * Standard_LRS: HDD
-	// * StandardSSD_LRS: Standard SSD
-	// * Premium_LRS: Premium SDD
-	// * UltraSSD_LRS: Ultra SDD
-	//
-	// Defaults to Premium_LRS. For more details, visit the Azure documentation:
-	// https://docs.microsoft.com/en-us/azure/virtual-machines/disks-types#disk-type-comparison
+	// diskStorageAccountType is the disk storage account type to use.
+	// Valid values are Standard, StandardSSD, PremiumSSD and UltraSSD and omitted.
+	// Note that Standard means a HDD.
+	// The disk performance is tied to the disk type, please refer to the Azure documentation for further details
+	// https://docs.microsoft.com/en-us/azure/virtual-machines/disks-types#disk-type-comparison.
+	// When omitted this means no opinion and the platform is left to choose a reasonable default, which is subject to change over time.
```

Comment:
For configuration APIs, default this on the controller side, and not the API side, else you have no way to change the default later should you wish to do so